obs-visca-control.lua

```diff
@@ -593,18 +660,37 @@ end
 function signal_on_activate(calldata)
     local source = obs.calldata_source(calldata, "source")
 	local settings = obs.obs_source_get_settings(source)
+	local source_name = obs.obs_source_get_name(source)
 
-    local do_preset = false
+    local do_action = false
     local active = obs.obs_data_get_int(settings, "scene_active")
     if (active == action_active.Program) or (active == action_active.Always) then
-        do_preset = true
+        do_action = true
     end
 
-    if do_preset then
-        do_cam_scene_action(settings)
+    if do_action then
+        do_cam_scene_action(settings, source_name)
     end
 
 	obs.obs_data_release(settings)
 end
 
+function signal_on_deactivate(calldata)
+    local source = obs.calldata_source(calldata, "source")
+	local source_name = obs.obs_source_get_name(source)
+    for a, data in pairs(previous_actions) do
+        -- log("pair %d action %s %s", a, data.source_name or 0, source_name)
+        if data.source_name == source_name and data.scene_action == actions.Animate then
+            if previous_actions[a+1] == nil then  -- only stop if there is no newer action
+                log("Camera stopping action of %s with action type %d", source_name, data.scene_action or 0)
+                do_cam_action_stop(data.camera_id, data.scene_action)
+                -- table.remove(previous_actions, a)
```

Comment:
Besides the fact that the use of a table for `previous_actions` is probably not needed, removal of the executed entry is correct. The check at line 684 however is not the right way; instead, the table should be iterated in reverse at line 681 (rendering line 684 useless), starting with the stop command for the last executed command first. By doing that, you are sure that a later added command is never accidentally stopped. 

Most probably you can get away with checking whether this scene has an action and if so, send the corresponding stop action. If it already stopped, it does no harm. That's a lot easier than global bookkeeping.